go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -539,21 +539,44 @@ func (p DoltDatabaseProvider) DropDatabase(ctx *sql.Context, name string) error
 	dbKey := formatDbMapKeyName(name)
 	db := p.databases[dbKey]
 
-	// Get the DB's directory
-	exists, isDir := p.fs.Exists(db.Name())
-	if !exists {
-		// engine should already protect against this
+	// get location of database that's being dropped
+	dbLoc := p.dbLocations[dbKey]
+	if dbLoc == nil {
 		return sql.ErrDatabaseNotFound.New(db.Name())
-	} else if !isDir {
-		return fmt.Errorf("unexpected error: %s exists but is not a directory", dbKey)
 	}
-
-	err = p.fs.Delete(db.Name(), true)
+	dropDbLoc, err := dbLoc.Abs("")
+	if err != nil {
+		return err
+	}
+	currentDbLoc, err := p.fs.Abs("")
```

Comment:
`currentDbLoc` could be a little misleading of a name. It makes me think of the current database a user has selected, but I don't think that's what we mean here. 

How about something like `rootDbLoc` or something similar to convey this is the root directory where all dbs are stored?